app/controllers/amazon_lookups_controller.rb

```diff
@@ -0,0 +1,51 @@
+class AmazonLookupsController < ApplicationController
+  def create
+    if is_empty?(gift_idea_params)
+      redirect_with_error
+    else
+      redirect_to new_list_person_gift_idea_path(
+        list,
+        person,
+        gift_idea: gift_idea_params,
+        amazon_lookup_complete: true
+      )
+    end
+  rescue
```

Comment:
shouldn't the if make the rescue unnecessary? How would it get past that if clause and still need the rescue?